gradle-jdks/src/main/java/com/palantir/gradle/jdks/GradleAwareCertificateSource.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks;
+
+import com.palantir.gradle.jdks.setup.CertificateSource;
+import com.palantir.gradle.jdks.setup.ILogger;
+import com.palantir.platform.GradleOperatingSystem;
+import com.palantir.platform.OperatingSystem;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.stream.Stream;
+import javax.inject.Inject;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderFactory;
+import org.gradle.process.ExecOutput;
+import org.gradle.process.ExecOutput.StandardStreamContent;
+
+public abstract class GradleAwareCertificateSource {
+
+    @Inject
+    protected abstract ProviderFactory getProviderFactory();
+
+    private Provider<OperatingSystem> osProvider;
+
+    private final Provider<byte[]> macosSystemCert = macosSystemCertificates();
+    private final Provider<byte[]> linuxSystemCert = linuxSystemCertificates();
+
+    @Inject
+    public GradleAwareCertificateSource(GradleOperatingSystem gradleOperatingSystem) {
```

Comment:
> FLUP replace this with OperatingSystem / GradleOperationSystem from utils

@FinlayRJW as per your suggestion, I've brought in GradleOperatingSystem here. 

As a FLUP, let's bump gradle-utils when the [fix](https://github.com/palantir/gradle-utils/pull/387) is out.